File: 算法基础课(修订版)/循环/P14074/verify_acm.py
class Solution:
    def solve(self, n):
        count = 0
        for i in range(1, n + 1):
            digit_sum = sum(int(ch) for ch in str(i))
            if digit_sum % 10 == i % 10:
                count += 1
        return count


if __name__ == "__main__":
    n = int(input())
    solution = Solution()
    print(solution.solve(n))

File: 算法基础课(修订版)/循环/P14074/verify_acm.cc
#include <iostream>
using namespace std;

class Solution {
public:
    int solve(int n) {
        int count = 0;
        for (int i = 1; i <= n; ++i) {
            int sumOfDigits = 0;
            int num = i;
            while (num > 0) {
                sumOfDigits += num % 10;
                num /= 10;
            }
            if (sumOfDigits % 10 == i % 10) {
                ++count;
            }
        }
        return count;
    }
};

int main() {
    int n;
    cin >> n;

    Solution solution;
    cout << solution.solve(n) << endl;

    return 0;
}
